packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -0,0 +1,619 @@
+        // No need to be exhaustive here
+        val emptyDictionary = emptyList<Pair<String, Int>>().toRealmDictionary()
+        assertTrue(emptyDictionary.isEmpty())
+
+        val elem1 = "A" to 1
+        val elem2 = "B" to 2
+
+        // We can create a dictionary from a list of one element...
+        val oneElementList = listOf(elem1)
+        oneElementList.toRealmDictionary()
+            .also { dictionaryFromSingleElementList ->
+                assertEquals(elem1.second, dictionaryFromSingleElementList[elem1.first])
+            }
+
+        // ... or from a list with many elements...
+        val multipleElementList = listOf(elem1, elem2)
+        multipleElementList.toRealmDictionary()
+            .forEach { assertTrue(multipleElementList.contains(Pair(it.key, it.value))) }
+
+        // ... or from a RealmMapEntrySet
+        // TODO will be added when support for dictionary.entries is added
+    }
+
+    @Test
+    fun nestedObjectTest() {
+        // TODO
+    }
+
+    @Test
+    fun copyToRealm() {
+        for (tester in managedTesters) {
+            tester.copyToRealm()
+        }
+    }
+
+    @Test
+    fun put() {
+        for (tester in managedTesters) {
+            tester.put()
+        }
+    }
+
+    @Test
+    fun get() {
+        for (tester in managedTesters) {
+            tester.get()
+        }
+    }
+
+    @Test
+    fun clear() {
+        for (tester in managedTesters) {
+            tester.clear()
+        }
+    }
+
+    @Test
+    fun entries_size() {
+        // TODO
+    }
+
+    @Test
+    fun entries_add() {
+        // TODO
+    }
+
+    @Test
+    fun entries_addAll() {
+        // TODO
+    }
+
+    @Test
+    fun entries_clear() {
+        // TODO
+    }
+
+    @Test
+    fun entries_iteratorNext() {
+        // TODO
+    }
+
+    @Test
+    fun entries_iteratorRemove() {
+        // TODO
+    }
+
+    @Test
+    fun entries_remove() {
+        // TODO
+    }
+
+    @Test
+    fun entries_removeAll() {
+        // TODO
+    }
+
+    // TODO missing in the C-API: https://github.com/realm/realm-core/issues/6181
+    @Test
+    fun containsKey() {
+        // TODO
+    }
+
+    // TODO missing in the C-API: https://github.com/realm/realm-core/issues/6181
+    @Test
+    fun containsValue() {
+        // TODO
+    }
+
+    private fun getTypeSafety(
+        classifier: KClassifier,
+        nullable: Boolean
+    ): DictionaryTypeSafetyManager<*> = when (nullable) {
+        true -> DictionaryTypeSafetyManager(
+            property = RealmDictionaryContainer.nullableProperties[classifier]!!,
+            dataSetToLoad = getDataSetForClassifier(classifier, true)
+        )
+        false -> DictionaryTypeSafetyManager(
+            property = RealmDictionaryContainer.nonNullableProperties[classifier]!!,
+            dataSetToLoad = getDataSetForClassifier(classifier, false)
+        )
+    }
+
+    @Suppress("UNCHECKED_CAST", "ComplexMethod")
+    private fun <T> getDataSetForClassifier(
```

Comment:
Also missing Decimal128 from this list, but I guess it hasn't been merged to `main` yet

Reply:
After you merged master in, I think you can remove the TODO again.